contracts/evm/contracts/adapters/CentralizedConnection.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+pragma abicoder v2;
+
+import "openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
+import "@xcall/utils/Types.sol";
+import "@xcall/contracts/xcall/interfaces/IConnection.sol";
+import "@iconfoundation/btp2-solidity-library/interfaces/ICallService.sol";
+
+
+contract XCallCentralizeConnection is Initializable, IConnection {
+
+    mapping(string => uint256) private messageFees;
+    mapping(string => uint256) private responseFees;
+
+    mapping(bytes32 => bool) public seenDeliveryVaaHashes;
+    address private xCall;
+    address private adminAddress;
+
+
+    event Message(string targetNetwork,int256 sn,bytes msg);
+
+
+    modifier onlyAdmin() {
+        require(msg.sender == this.admin(), "OnlyAdmin");
+        _;
+    }
+
+    function initialize(address _xCall ) public initializer {
+        adminAddress = msg.sender;
+        xCall = _xCall;
+    }
+
+
+    function setFee(
+        string calldata networkId,
+        uint256 messageFee,
+        uint256 responseFee
+    ) external onlyAdmin {
+        messageFees[networkId] = messageFee;
+        responseFees[networkId] = responseFee;
+    }
+
+
+    function getFee(string memory _to, bool _response) external view override returns (uint256 _fee) {
+        uint256 messageFee = messageFees[_to];
+        if (_response == true) {
+            uint256 responseFee = responseFees[_to];
+            return messageFee + responseFee;
+        }
+        return messageFee;
+    }
+
+
+    function sendMessage(
+        string calldata _to,
+        string calldata _svc,
+        int256 _sn,
+        bytes calldata _msg
+    ) external override payable {
+        require(msg.sender == xCall, "Only xCall can send messages");
+        uint256 fee = this.getFee(_to,false);
+        require(msg.value >= fee,"Fee is not Sufficient"); 
```

Comment:
Should be true if sn > 0
If sn == 0 should bne true
If sn < 0 no fee is required

Reply:
On second thought we can remove fee checks, xCall will verify this for us to save on gas

Reply:
yes, removed fee checks from connection!

Reply:
For every adapters, we have this check on adapter itself, so maintaining consistency, I added check on connection contract!